R/mlr_graphs_elements.R

```diff
@@ -126,28 +127,29 @@ mlr_graphs$add("robustify", pipeline_robustify)
 #'   replicated and subsampled graph's.
 #'   In the simplest case, `po("classifavg")` and `po("regravg")` can be used
 #'   in order to perform simple averaging of classification and regression
-#'   predictions respectively.`
+#'   predictions respectively.
 #'   If `NULL` (default), no averager is added to the end of the graph.
+#'   Note that setting `collect_multipliciy = TRUE` during construction of the averager is required.
 #' @return [`Graph`]
 #' @export
 #' @examples
 #' \donttest{
 #' library(mlr3)
 #' lrn_po = po("learner", lrn("regr.rpart"))
 #' task = mlr_tasks$get("boston_housing")
-#' gr = pipeline_bagging(lrn_po, 3, averager = po("regravg"))
+#' gr = pipeline_bagging(lrn_po, 3, averager = po("regravg", collect_multiplicity = TRUE))
 #' resample(task, GraphLearner$new(gr), rsmp("holdout"))
 #' }
 pipeline_bagging = function(graph, iterations = 10, frac = 0.7, averager = NULL) {
+  g = as_graph(graph, clone = TRUE)
   assert_count(iterations)
   assert_number(frac, lower = 0, upper = 1)
-  graph = as_graph(graph)
-  if (!is.null(averager)) averager = as_graph(averager)
-
-  subs = po("subsample", param_vals = list(frac = frac)) %>>% graph
-  subs_repls = pipeline_greplicate(subs, iterations)
+  if (!is.null(averager)) averager = as_graph(averager, clone = TRUE)
```

Comment:
you should add a test that the averager does collect multiplicities (length of `input` is 1 and its train and predict type both start with a `[`).

Reply:
also test that creating a bagging pipeline with a pipeop that does not collect multiplicities throws an error.

Reply:
yes, makes sense! thanks for the heads-up!